src/torchcodec/decoders/_video_decoder.py

```diff
@@ -180,13 +222,42 @@ def __init__(
             custom_frame_mappings=custom_frame_mappings_data,
         )
 
+        self._fallback_info = FallbackInfo()
+        self._has_decoded_frame = False
```

Comment:
Instead of tracking whether a frame has been decoded, we could also just call `_update_cpu_fallback` inside every method that would decode a frame. The problem with that approach would be that we would be returning non tensors in the compiled methods. As such, we would have to do something like inside 
```
if torch.compiler.is_compiling():
    return
```

Reply:
I think we should implement this `_has_decoded_frame` logic in C++ for 2 reasons:

- it's simpler and can be localized to a single place (see suggested implementation below)
- it's only relevant for the default interface, not for the beta interface.

So maybe all we need is to update the default interface's returned string, and let it indicate whether its status is known or unknown yet. Here:

https://github.com/meta-pytorch/torchcodec/blob/38fa96c6226ce502c0f86fffa664835059d35ce9/src/torchcodec/_core/CudaDeviceInterface.cpp#L357-L363


To know whether a frame has been decoded yet, I think we can simply set a boolean field to true when 

https://github.com/meta-pytorch/torchcodec/blob/38fa96c6226ce502c0f86fffa664835059d35ce9/src/torchcodec/_core/CudaDeviceInterface.cpp#L238

is called.